src/tools/rustbook/src/main.rs

```diff
@@ -8,10 +8,9 @@ use clap::{App, ArgMatches, SubCommand, AppSettings};
 use mdbook::MDBook;
 use mdbook::errors::{Result as Result3};
 
-#[cfg(all(target_arch = "x86_64", target_os = "linux"))]
+#[cfg(feature = "linkcheck")]
```

Comment:
I don't think this interacts with the build script in the way that you're thinking, the `linkcheck` directive in the build script doesn't activate this feature.

Reply:
I'm not sure how to trigger linkchecker error but from Cargo book:
> rustc-cfg=FEATURE indicates that the specified feature will be passed as a --cfg flag to the compiler. This is often useful for performing compile-time detection of various features.
https://doc.rust-lang.org/cargo/reference/build-scripts.html

And rustc book:
> --cfg: configure the compilation environment
This flag can turn on or off various #[cfg] settings.
https://doc.rust-lang.org/rustc/command-line-arguments.html#a--cfg-configure-the-compilation-environment

Am I missing something?

Reply:
Er well... no it's not how it works. I'd recommend playing around with this locally and try to get it to execute. There's a difference between `--cfg linkcheck` and `--cfg feature="linkcheck"`